### PR TITLE
gecko: Updates for hwmv2

### PR DESCRIPTION
**Pull request by nordicjm**
Adds updates for hwmv2

### CODE REVIEW

File: gecko/CMakeLists.txt [hunk 1]
```diff
@@ -16,7 +16,7 @@ string(SUBSTRING ${CONFIG_SOC_SERIES} 7 2 GECKO_SERIES_NUMBER)
 
 set(SILABS_GECKO_PART_NUMBER ${CONFIG_SOC_PART_NUMBER})
 
-if (${CONFIG_BOARD} STREQUAL "efr32_radio_brd4187c")
+if("${CONFIG_BOARD}" STREQUAL "efr32_radio")
   set(PRODUCT_NO "sdid215")
```
Comment (jerome-pouiller) [inline gecko/CMakeLists.txt:20]: `PRODUCT_NO` should be set to `sdid215` only for `brd4187c`. In fact `PRODUCT_NO` only depends on the SOC:
```
efm32wg    | sdid74
efm32hg    | sdid77
efm32pg1b  | sdid80
efr32fg1p  | sdid80
efm32jg12b | sdid84
efm32pg12b | sdid84
efr32mg12p | sdid84
efr32bg13p | sdid89
efr32fg13p | sdid89
efm32gg11b | sdid100
wgm160     | sdid100
efm32gg12b | sdid106
efr32mg21  | sdid200
efr32bg22  | sdid205
efr32mg24  | sdid215
efr32bg27  | sdid230
```

File: gecko/CMakeLists.txt [hunk 2]
```diff
@@ -25,14 +25,14 @@ if (${CONFIG_BOARD} STREQUAL "efr32xg24_dk2601b")
...
-
-if (${CONFIG_BOARD} STREQUAL "efr32bg22_brd4184b")
-  set(SILABS_GECKO_BOARD "brd4184b")
-  set(PRODUCT_NO "sdid205")
+if("${CONFIG_BOARD}" STREQUAL "efr32bg22_brd4184")
+  if("${CONFIG_BOARD_REVISION}" STREQUAL "A")
+    set(SILABS_GECKO_BOARD "brd4184a")
+    set(PRODUCT_NO "sdid205")
+  elseif("${CONFIG_BOARD_REVISION}" STREQUAL "B")
+    set(SILABS_GECKO_BOARD "brd4184b")
```
Comment (jerome-pouiller) [inline gecko/CMakeLists.txt:33]: In fact, I believe we should keep `efr32bg22_brd4184a` and `efr32bg22_brd4184b` as two separated boards and get rid of `CONFIG_BOARD_REVISION`.

For information, the full names of the boards with revision are `brd4184a rev A01`, `brd4184a rev A02`, `brd4184b rev A01`, etc... But until now, we didn't have to change any configuration between the revisions.

Comment (nordicjm) [reply gecko/CMakeLists.txt:34]: Fixed

File: gecko/CMakeLists.txt (hunk 1 above)
Comment (nordicjm) [reply gecko/CMakeLists.txt:20]: Fixed

**Review by fkokosinski — APPROVED** [review]

**Comment by fkokosinski** [comment]
Hey @nordicjm, what is the plan here? I assume we'll want to keep off from merging HWMv2-related stuff in modules until the `collab-hwm` branch is ready, because the associated Zephyr PR was merged with `west.yml` pointing to this PR. Plus https://github.com/zephyrproject-rtos/hal_silabs/pull/59 kind of supersedes this PR.

Please let me know if/when some action from me is needed here

**Comment by nordicjm** [comment]
> Hey @nordicjm, what is the plan here? I assume we'll want to keep off from merging HWMv2-related stuff in modules until the `collab-hwm` branch is ready, because the associated Zephyr PR was merged with `west.yml` pointing to this PR. Plus #59 kind of supersedes this PR.
> 
> Please let me know if/when some action from me is needed here

Will need to merge them once the main PR is opened and approved from collab-hwm to zephyr

**Pull request closed without merging**